app/controllers/email_signins_controller.rb

```diff
@@ -15,11 +15,8 @@ def add
       expires_at: expires_at
     )
 
-    if magic_link.save
-      return render json: { token: magic_link.id }, status: :created
-    else
-      return render_email_missing_error
```

Comment:
I considered writing a test for this scenario, but then I realised this will be a very rare one as there is no primary key in the table thus we can create as many magic links as we want with the same attributes, without failure. The only way the save could fail (return false) is if there is a DB connectivity blip, in which case we probably don't want to return with "render_email_missing_error" but with something more appropriate like server error.

Thus I remove this if-else and changed the `#save` to `#save!` which will raise an exception in case the save fails wichi will be covered by the generic error handling concern returning a more appropriate error, and more importantly reporting it to Sentry.